### ISSUE TITLE
Support i686?

### ISSUE THREAD

**Issue by roptat** [Ekdohibs/camlboot#52]
Since I pushed camlboot to Guix, we kinda need to support the platform it supports (well, I can also fix that by not using camlboot on i686). Here is the error I get (see https://ci.guix.gnu.org/build/18557/log/raw):

```
mkdir -p lex/
../../ocaml-src/byterun/ocamlrun lex.boot.byte ../../ocaml-src/lex/lexer.mll -o lex/lexer.ml
Fatal error: exception Failure("input_value: object too large to be read back on a 32-bit platform")
make[1]: *** [Makefile:28: lex/lexer.ml] Error 2
```

Is this an issue with `lex.boot.byte` (because we maybe only create bytecode for 64 bit architectures), or with `lexer.mll` (because maybe it creates huge objects)?

**Comment by Ekdohibs** [comment]
The issue is in `compile.scm`, where serialization only targets 64-bit platforms. I think this can be changed (see `runtime/extern.c` in ocaml sources), but I will need some time before working on it.

**Comment by gasche** [comment]
That sounded fun (?!), so I tried to implement the suggested fix in #53. @roptat, would you be able to test the branch with your 32bit system?

**Comment by Millak** [comment]
To bring some closure to this question, Guix does now have substitutes available for i686-linux, so it was a success.

Thanks!
